errors/errors.go

```diff
@@ -32,7 +32,8 @@ func (e Error) Error() string {
 // New("some error") returns error with exact string passed
 // New("some error: %v", "some context") returns error with message "some error: some context"
 func New(format string, args ...interface{}) error {
-	return Error(fmt.Sprintf(format, args...))
+	err := Error(fmt.Sprintf(format, args...))
```

Comment:
could you make a bench mark test for an error with and without stack trace? I'm curious about the performance impact of getting the stack trace each time this is called. If the cost is too high I would prefer to have a way to switch it off. Other than that LGTM

Reply:
Each error would have around 56 bytes added to them. Since these will be created only at times we need to raise an error, this is an acceptable cost

Reply:
I don't mind about the storage, its not a concern at all.